### PR TITLE
feat(vector): Upgrade chart to Vector 0.26.0

### PR DESCRIPTION
**Pull request by jszwedko**
Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: charts/vector/ci/vector-templating-values.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ customConfig:
       inputs: [generator]
       bucket: logs-archive
       key_prefix: >-
-        {{ print "{{kubernetes.pod_labels.app\\.kubernetes\\.io/client-id}}/%Y/%m/%d/{{kubernetes.pod_name}}/" }}
```
Comment (jszwedko) [inline charts/vector/ci/vector-templating-values.yaml:20]: This seems to have been making it through to Vector as `\\.`s.

Comment (spencergilbert) [reply charts/vector/ci/vector-templating-values.yaml:20]: :D You mean we should follow the docs I recently updated 🙃 

https://vector.dev/docs/reference/configuration/template-syntax/

Comment (jszwedko) [reply charts/vector/ci/vector-templating-values.yaml:20]: Yes indeed 😄 I made the quoting changes in [a82cde9](https://github.com/vectordotdev/helm-charts/pull/275/commits/a82cde9f77b96727ba6a4f07078c27f1fad979ed)

**Review by spencergilbert — APPROVED** [review]
lgtm, happy to look at it myself locally if you'd like

**Pull request merged**